airflow/models/dagrun.py

```diff
@@ -1183,6 +1183,14 @@ def _find_missing_task_indexes(
             existing_indexes[task].append(ti.map_index)
             task.run_time_mapped_ti_count.cache_clear()  # type: ignore[attr-defined]
             new_length = task.run_time_mapped_ti_count(self.run_id, session=session) or 0
+
+            if ti.map_index >= new_length and new_length > 0:
```

Comment:
Why do we need the `new_length > 0` check here?

Reply:
Yeah, not needed. I was taking clue from _check_for_removed_or_restored_tasks but I see now that it's not needed